util.c

```diff
@@ -5322,3 +5286,159 @@ int util_is_wlan(char *device)
   return slist_getentry(config.net.wlan.devices, device) ? 1 : 0;
 }
 
+
+/*
+ * Write log message.
+ *
+ * level is a bitmask determining the destination to log to.
+ *
+ * Add a time stamp when the destination has the LOG_TIMESTAMP flag set.
+ */
+void util_log(unsigned level, char *format, ...)
+{
+  va_list args;
+  char *buf, *caller = NULL;
+  int buf_len = 0;
+  log_file_t *lf;
+
+  time_t t = time(NULL);
+  struct tm *gm = gmtime(&t);
+
+  va_start(args, format);
+  if(vasprintf(&buf, format, args) == -1) buf = NULL;
+  if(buf) buf_len = strlen(buf);
+  va_end(args);
+
+  for(lf = config.log.dest; lf < config.log.dest + sizeof config.log.dest / sizeof *config.log.dest; lf++) {
+    if((level & lf->level)) {
```

Comment:
I'm aware of that and done this numerous times in the linuxrc code. So you might also have learned from me... ;-)